packages/amplify-category-storage/provider-utils/awscloudformation/service-walkthroughs/dynamoDb-walkthrough.js

```diff
@@ -166,33 +181,33 @@ async function configure(context, defaultValuesFilename, serviceMetadata, resour
   while (continueAttributeQuestion) {
     const attributeAnswer = await inquirer.prompt([attributeQuestion, attributeTypeQuestion]);
 
-    if (attributeAnswers.findIndex(attribute => attribute.AttributeName === attributeAnswer[inputs[2].key]) !== -1) {
+    if (attributeAnswers.findIndex(attribute => attribute.AttributeName === attributeAnswer[inputs[3].key]) !== -1) {
       continueAttributeQuestion = await amplify.confirmPrompt.run(
-        'This attribute was already added. Do you want to add another attribute?'
+        'This attribute was already added. Do you want to add another attribute?',
       );
       continue;
     }
 
     attributeAnswers.push({
-      AttributeName: attributeAnswer[inputs[2].key],
```

Comment:
I'd love to not have to have these indexed by numbers how they are, as it's a bit easy to break. The theory here is that I've added to `packages/amplify-category-storage/provider-utils/supported-services.json` at position [2], and so everything after that has been incremented by 1. Hopefully I got them all, but it's tricky as there aren't any tests. I will be revisiting this at some point to see if I can get rid of those indexes and write some tests, for this PR though, it may be worth checking that every index >=2 has been incremented.